packages/flutter/lib/src/material/outline_button.dart

```diff
@@ -354,22 +354,32 @@ class _OutlineButtonState extends State<OutlineButton> with SingleTickerProvider
     return Colors.transparent;
   }
 
+  /// The button outline has a color and a width.
+  ///
+  /// If [this.borderSize] is provided and the style is none, then that is used.
+  ///
+  /// Otherwise, both the color and width will attempt to assign the most
+  /// specific provided value. If there is no provided value, then the theme
+  /// value is used. If te theme value is not provided, then the baseline
+  /// material value is used.
   BorderSide _getOutline(ThemeData theme, ButtonThemeData buttonTheme) {
-    final bool themeIsDark = theme.brightness == Brightness.dark;
     if (widget.borderSide?.style == BorderStyle.none)
       return widget.borderSide;
 
+    // TODO(antrob): Centralize this default once surfaceColor exists.
+    final Color baselineSurfaceColor = theme.brightness == Brightness.dark
+        ? Colors.white : Colors.black;
+    // TODO(antrob): Replace primaryColor with surfaceColor.
+    final Color themeColor = baselineSurfaceColor ?? theme.primaryColor;
     final Color color = widget.enabled
       ? (_pressed
-         ? widget.highlightedBorderColor ?? theme.primaryColor
-         : (widget.borderSide?.color ??
-            (themeIsDark ? Colors.grey[600] : Colors.grey[200])))
-      : (widget.disabledBorderColor ??
-         (themeIsDark ? Colors.grey[800] : Colors.grey[100]));
+          ? widget.highlightedBorderColor ?? themeColor
+          : (widget.borderSide?.color ?? themeColor))
+      : (widget.disabledBorderColor ?? themeColor);
```

Comment:
I doubt we ever want to use the theme's primary color for the disabled border color. If the current grey fallbacks aren't correct, we should figure out what the Material defaults should be.